src/TLuaInterpreter.cpp

```diff
@@ -10147,6 +10147,18 @@ int TLuaInterpreter::getTime(lua_State* L)
 }
 
 
+// syntax: getEpoch()
+// returns: seconds since unix epoch with milliseconds (e.g. 1523555867.191)
+int TLuaInterpreter::getEpoch(lua_State *L)
+{
+    QDateTime time = QDateTime::currentDateTime();
+
+    lua_pushnumber(L, double(time.toMSecsSinceEpoch()) / 1000.0);
```

Comment:
We are writing C++ code so can we have a C++ `static_cast<double>(value)` - please... oh, actually the (C style) cast is redundant, you are (correctly!) dividing by a double and that will automatically cast the result to a double anyhow.  

Ideally we'd truncate the number of digits after the decimal point but that would need a bit more jiggery-pokery to achieve.

Reply:
C++ is not my forte. Pardon if I don't fully understand what you are asking. Used double() because it was returning qint64 (toMSecsSinceEpoch method returns type qint64) instead of a floating point - even when dividing by a double. Please let me know how to correct this if needed.

Reply:
@SlySven does anything need to be done here?

Reply:
Well I do not believe the result was actually an integer - the rules on conversion are well defined (apparently - see [here](https://stackoverflow.com/a/13663723/4805858) for instance) so the division by a double (`1000.0`) will convert the result to that type as well.  The only thing I can think of is that whatever test values you used it happened to divide cleanly into an integer...

If it is not behaving as it should I guess you could replace both lines with:

    lua_pushnumber(L, static_cast<double>(QDateTime::currentDateTime().toMSecsSinceEpoch() / 1000.0));

however that cast should not be needed according to the stuff I know... 😕

Reply:
In that SO, it was mentioned that *usually* integer-like numbers are promoted to to floating-point equivalents. But things are complicated by size differences and sign. Is it possible that - without the cast - promotion was not taking place due to the size of the return type of toMSecsSinceEpoch? (qint64)

Reply:
Humm, I've tested this and I get the right results without any casting and using `QDateTime::currentDateTime().toMSecsSinceEpoch() / 1000.0` as the second argument to the `lua_pushnumber(`...`)` function directly... 😁

Should make timing longer intervals (> 24 Hours) easy too!

Reply:
Ok, if you are happy to put in the C++ style `static_cast<`class name`>(`value to cast`)` - I'll be okay with it - and you can eliminate the intermedia `time` and use the `QDateTime::toMSecsSinceEpoch()` method on the return value from the `static` method `QDateTime::currentDateTime()` {Amonsgt other things, *static* means it works directly without having to create a `QDateTime` instance, like `time` to use it on}

Reply:
@salcedo mind doing this small tweak?